testing/generate.js

```diff
@@ -53,13 +73,34 @@ function playwrightTest(appName, testNames, folderName) {
   const encodedFolderName = encodeURIComponent(folderName);
   const testAppName = appName.replace("'", "");
 
+  const beforeEachHook =
+`  test.beforeEach(async ({ page }) => {
+    try {
+      if (!fs.existsSync(resultsPath)) {
+        const app = new RetoolApplication(page, "${encodedAppName}", "${folderName ? encodedFolderName : ''}")
+        const results = await app.test()
+
+        fs.writeFileSync(resultsPath, results)
+      }
+    } catch(err) {
+      console.error(err)
+    }
+  })`
+
   const individualTests = testNames.map(test =>
-`  test('${test}', async ({ page }) => {
-    const app = new RetoolApplication(page, "${encodedAppName}", "${folderName ? encodedFolderName : ''}")
-    await app.test('${test}')
+`  test('${test}', async () => {
+    if (fs.existsSync(resultsPath)) {
+      const rawResults = fs.readFileSync(resultsPath)
+      const results = JSON.parse(rawResults.toString())
+      expect(results['${test}']).toBe(true)
+    } else {
+      console.error(resultsPath + " does not exist")
```

Comment:
A better way to write this is to invert the conditional, as it results is less indentation.

```js
if (!fs.existsSync(resultsPath)) {
  throw new Error("test-results do not exist")
}
const rawResults = fs.readFileSync(resultsPath)
const results = JSON.parse(rawResults.toString())
expect(results['${test}']).toBe(true)
```